### PR TITLE
[added] Night ambient music

### PR DESCRIPTION
**Pull request by mugg91**
## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

Fixes 2) of #1295.

This adds a new night world ambient music track to WARMusic.as and ChallengeMusic.as.
It is crickets chirping which is common to occur at late afternoon to midnight.
This music plays when `getMap().getDayTime()` is between 0.75 and 1.0 - before sunset starts until before dawn starts.

The music used is the first 45 seconds of this, amplied by +6.1 and saved as .ogg at quality level 6. It is free-to-use: 
https://pixabay.com/sound-effects/cricket-sound-14867/
If you think there is a better soundtrack, let me know.

## Steps to Test or Reproduce

Go to Sandbox or TTH tutorial.
Wait until sunset and night (or type `getMap().SetDayTime(0.7f);` in console to jump right before a sunset).
Notice the night ambient music.

### CODE REVIEW

**Review by AsuMagic — CHANGES_REQUESTED** [review]
The background music really doesn't fit, you can easily hear city sounds and cars running. :P

**Comment by mugg91** [comment]
> The background music really doesn't fit, you can easily hear city sounds and cars running. :P

Please note, only the first 45 seconds are used. The part of the car driving is not included.
Any other sound effects are basicly inaudible in actual gameplay (note, there is the game music running on top of this) and could be assumed to be village sounds or wind.

File: Entities/Meta/ChallengeMusic.as [hunk 1]
```diff
@@ -124,7 +134,10 @@ void GameMusicLogic(CBlob@ this, CMixer@ mixer)
 			if (timer % 24 != 0)
 				return;
 
-			if (mixer.isPlaying(world_ambient) || mixer.isPlaying(world_ambient_underground) || mixer.isPlaying(world_ambient_mountain))
+			if (mixer.isPlaying(world_ambient) 
+				|| mixer.isPlaying(world_ambient_underground) 
+				|| mixer.isPlaying(world_ambient_mountain
```
Comment (AsuMagic) [inline Entities/Meta/ChallengeMusic.as:139]: Missing `)`, this causes a syntax error.

File: Entities/Meta/ChallengeMusic.as [hunk 2]
```diff
@@ -95,15 +97,23 @@ void GameMusicLogic(CBlob@ this, CMixer@ mixer)
 		if (timer % 48 != 0)
 			return;
 
+		bool isNight = map.getDayTime() > 0.75f;
```
Comment (AsuMagic) [inline Entities/Meta/ChallengeMusic.as:100]: This seems to be a generally poor value. `>0.85 && <0.1` would probably be more fitting with the current sky gradient.

**Review by AsuMagic — CHANGES_REQUESTED** [review]

File: Entities/Meta/ChallengeMusic.as (hunk 2 above)
Comment (AsuMagic) [reply Entities/Meta/ChallengeMusic.as:100]:
```suggestion
		bool isNight = map.getDayTime() > 0.85f && map.getDayTime() < 0.1f;
```
I have now realized that fix suggestions are a thing.

File: Entities/Meta/ChallengeMusic.as (hunk 1 above)
Comment (AsuMagic) [reply Entities/Meta/ChallengeMusic.as:139]:
```suggestion
				|| mixer.isPlaying(world_ambient_mountain)
```

**Review by AsuMagic — DISMISSED** [review]

**Review by AsuMagic — APPROVED** [review]

**Pull request merged**